cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/mazha/CouchConfig.java

```diff
@@ -34,33 +39,98 @@
     // a reverse proxy
     private URI rootUri;
 
+    private String username;
+
+    private String password;
+
     private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
             <HttpConnectionRequestInterceptor>();
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
     public CouchConfig(URI rootUri) {
-        this.rootUri = rootUri;
+        this(rootUri,
+                Collections.<HttpConnectionRequestInterceptor>emptyList(),
+                Collections.<HttpConnectionResponseInterceptor>emptyList(),
+                null,
+                null);
     }
 
     public CouchConfig(URI rootUri,
                        List<HttpConnectionRequestInterceptor> requestInterceptors,
-                       List<HttpConnectionResponseInterceptor> responseInterceptors) {
+                       List<HttpConnectionResponseInterceptor> responseInterceptors,
+                       String username,
+                       String password) {
         this.rootUri = rootUri;
         this.requestInterceptors = requestInterceptors;
         this.responseInterceptors = responseInterceptors;
+        this.username = username;
+        this.password = password;
+
     }
 
-    public List<HttpConnectionRequestInterceptor> getRequestInterceptors() {
+
+    public List<HttpConnectionRequestInterceptor> getRequestInterceptors(boolean includeCookie) {
+        CookieInterceptor cookieInterceptor = buildCookieInterceptor();
+        if (includeCookie && cookieInterceptor != null) {
+            List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
+                    <HttpConnectionRequestInterceptor>();
+            requestInterceptors.addAll(this.requestInterceptors);
+            requestInterceptors.add(cookieInterceptor);
+            return requestInterceptors;
+        }
         return requestInterceptors;
     }
 
+    public List<HttpConnectionResponseInterceptor> getResponseInterceptors(boolean includeCookie) {
+        CookieInterceptor cookieInterceptor = buildCookieInterceptor();
+        if (includeCookie && cookieInterceptor != null) {
+            List<HttpConnectionResponseInterceptor> requestInterceptors = new ArrayList
+                    <HttpConnectionResponseInterceptor>();
+            requestInterceptors.addAll(this.responseInterceptors);
+            requestInterceptors.add(cookieInterceptor);
+            return requestInterceptors;
+        }
+        return responseInterceptors;
+    }
+
+    private CookieInterceptor buildCookieInterceptor() {
```

Comment:
Its to make the number of code changes vastly smaller than otherwise. Through out the test code we pull authentication information out of the config, and it is expected that the cookie interceptor will be present. This could be fixed just for the replicator tests be removing all instances of the cookie interceptor in the interceptor lists. But I thought it would be a better decision to not include it from the config, but still provide the authentication mechanism to any of the other tests.